adareducer/project_support.py

```diff
@@ -1,32 +1,28 @@
-import subprocess
 import os
-from pathlib import Path
-import glob
+import libadalang as lal
 
 
 class ProjectResolver(object):
     """Utility to resolve base names"""
 
     def __init__(self, project_file):
         self.files = {}
-        # gprls doesn't work so use something else
-        dir = Path(project_file).resolve().parent
-        for adx in dir.glob(f"**/*.ad?"):
-            if not adx.name.startswith("b__"):
+
+        # Use the Libadalang API to query the sources for this project
+        files = lal.SourceFiles.for_project(project_file)
+
+        for full_path in files:
+            basename = os.path.basename(full_path)
```

Comment:
Why dropping pathlib?  This could be done with a `Path` and it's so more python3 :)

Reply:
I honestly don't care much about scoring high on the python3-o-meter but okay I can do it!

Reply:
Done in the next commit!